### DISCUSSION TITLE
Step-by-step guide installation

### DISCUSSION THREAD

**Discussion by budisek** [rednblkx/HomeKey-ESP32#25] [Ideas]
It would be good to have easy step-by-step Installation guide for not such skilled people like me.

**Comment by rednblkx** [comment]
That is something that will be worked on in the future as the code is still a work in progress and Guides would be subject to change frequently, so i've only added some surface knowledge so people that come across this repo know what it is about.

If you have any issues, you can let me know in this discussion thread or any other relevant discussion thread open.

**Comment by Joeknx** [comment]
A short instruction would be nice!
E.g. I have a ESP32 and a Mac with ESPTool. What do I do next to get the program to work on the ESP.
Thanks!

**Reply by rednblkx** [reply]
Hi There!
Sorry for the late reply.

First, you will need to install PlatformIO Core, you can find information about the installation [here](https://docs.platformio.org/en/latest/core/installation/methods/installer-script.html), you will need to have Python 3 installed, once it's installed run `platformio run -t upload`

If you are using Visual Studio Code you can download the PlatformIO IDE for VSCode extension, restart the editor, let it finish configuring the project and then press the Upload button(button with an arrow) on the bottom of the window.

I actually recommend using the vscode extension as the process is simpler and less prone to issues.

Let me know if you encounter any issues.

**Reply by Joeknx** [reply]
Visual Studio Code was a great hint! I made it and it works!
Thanks a lot!

**Comment by rednblkx** [comment]
Hi!

I know it may be late for the people on this thread, but just wanted to say that i've now created a wiki that i hope has easy to follow instructions for people to get started.

Would very much appreciate your feedback on this, let me know if there is something i should add/change/remove.

**Reply by electric-m** [reply]
I am sure, many users will appreciate that.
Maybe yo would want add another chapter for „upgrading“?
Questions might be:

- Just connect USB, rebuild and reupload?
- Changes for OTA update, adjusting the upload port to wifi instead of serial and so on…

**Comment by dekamilv** [comment]
Perfect ! 
Thanks a lot. 
Will try to compile next week.
